src/scripts/main.js

```diff
@@ -1,3 +1,71 @@
 'use strict';
 
 // write code here
+const table = document.querySelector('tbody');
+const buttonAddRow = document.querySelector('.append-row.button');
+const buttonAddColumn = document.querySelector('.append-column.button');
+const buttonRemoveColumn = document.querySelector('.remove-column.button');
+const buttonRemoveRow = document.querySelector('.remove-row.button');
+
+document.addEventListener('click', () => {
+  const columns = document.querySelector('tr').children.length;
+  const rows = document.querySelectorAll('tr').length;
+
+  if (columns === 10) {
+    buttonAddColumn.setAttribute('disabled', 'disabled');
+  } else {
+    buttonAddColumn.removeAttribute('disabled');
+  }
+
+  if (columns === 2) {
+    buttonRemoveColumn.setAttribute('disabled', 'disabled');
+  } else {
+    buttonRemoveColumn.removeAttribute('disabled');
+  }
+
+  if (rows === 10) {
+    buttonAddRow.setAttribute('disabled', 'disabled');
+  } else {
+    buttonAddRow.removeAttribute('disabled');
+  }
+
+  if (rows === 2) {
+    buttonRemoveRow.setAttribute('disabled', 'disabled');
+  } else {
+    buttonRemoveRow.removeAttribute('disabled');
+  }
+});
```

Comment:
The event listener is attached to the entire document, which means the logic for enabling/disabling buttons will run on every click anywhere on the page. This might not be efficient. Consider attaching the event listener to a more specific element or refactoring the logic to only run when necessary.